xdsl/dialects/builtin.py

```diff
@@ -1566,52 +1566,51 @@ class StridedLayoutAttr(MemRefLayoutAttr, BuiltinAttribute, ParametrizedAttribut
     An attribute representing a strided layout of a shaped type.
     See external [documentation](https://mlir.llvm.org/docs/Dialects/Builtin/#stridedlayoutattr).
 
-    Contrary to MLIR, we represent dynamic offsets and strides with
-    `NoneAttr`, and we do not restrict offsets and strides to 64-bits
```

Comment:
@math-fehr, what was the reason behind this?

Reply:
I personally felt that encoding the dynamic offset by just using a special number is bad practice, because it's really easy to use the wrong one (`-1` versus `INT_MIN` versus `INT_MAX`).
If you want to use MLIR version that's fine to me, but I personally prefer `NoneAttr` here unless there is some benefit to follow MLIR.

Reply:
I would personally prefer having a check that the values are non-negative then!
Also, having things like `NoneAttr` would technically allow us to write constraints that would check that there are no dynamic values for instance.

Reply:
We could do that with -1 also but I currently don't have a strong opinion :) I'l update this PR

Reply:
So why do you want to change it then?
My understanding is that this is equivalent to poison?